BSP/Simulator_V2/Src/BSP_Timer.c

```diff
@@ -50,6 +59,6 @@ uint32_t BSP_Timer_GetRunFreq(void) {
  * @return  Microseconds 
  */
 uint32_t BSP_Timer_GetMicrosElapsed(void) {
-    // TODO
-    return 42;
+    uint32_t MicrosElasped = (BSP_Timer_GetTicksElasped() * 1000000) / CLOCKS_PER_SEC;
```

Comment:
This is kind of nitpicky and won't make a difference here, but it might be better to call BSP_Timer_GetRunFreq instead of CLOCKS_PER_SECOND for consistency with the function in the BSP layer.

Reply:
True but BSP_Timer_GetRunFreq is only called in this file nowhere else making no difference when used in simulator. Also implementing getrunfreq is i don't think possible or at least not worth the effort as its only use is in the getmicroselapsed.